### ISSUE TITLE
Dependency on node types

### ISSUE THREAD

**Issue by bodograumann** [form-data/form-data#450]
I am working on https://github.com/OpenAPITools/openapi-generator/issues/802 and found that when updating to form-data@2.5 with included typings, I get errors:

```
node_modules/form-data/index.d.ts:6:23 - error TS2688: Cannot find type definition file for 'node'.
6 /// <reference types="node" />

node_modules/form-data/index.d.ts:7:25 - error TS2307: Cannot find module 'stream'.
7 import * as stream from 'stream';

node_modules/form-data/index.d.ts:8:23 - error TS2307: Cannot find module 'http'.
8 import * as http from 'http';
                        ~~~~~~
node_modules/form-data/index.d.ts:28:16 - error TS2304: Cannot find name 'Buffer'.
28   getBuffer(): Buffer;
                  ~~~~~~
```

… unless I install `@types/node` myself manually.
So I think you should move that dependency from devDependencies to dependencies.

**Comment by alexindigo** [comment]
`@types/node` is expected to be present if the project is using TypeScript and is targeting `node`, which is the main (if not only) target for `form-data`.

Curious how come that project doesn't have `@types/node` but uses `form-data`?

**Comment by bodograumann** [comment]
Yeah, I am not the original author of that code and I think we are just finding our way in terms of node vs. browser target.
Originally we were using `form-data@2.3.2`, where we had to explicitly depend on `@types/form-data`. At that point `@types/node` was included as a dependency of `@types/form-data` automatically. So only after the update we noticed that we actually need `@types/node` for our own code.

So I guess you have a good point that most typescript users of this library should already have `@types/node` installed.
Also afaik npm does not have a way to say “this dependency is only necessary with typescript“.
